project-codeflare/codeflare-sdk
# Added support for ingress over routes on cluster creation

# Issue link

Closes #219 
# What changes have been made

Added an ingressOptions dict for specifying what options you want to set for your dashboard ingress.
An example of the specified ingressOptions would look like this (based off the routes for dashboard)
```
cluster = Cluster(ClusterConfiguration(local_interactive=local_interactive, namespace=namespace, name=cluster_name, num_workers=1, min_cpus=1, max_cpus=1, min_memory=4, max_memory=4, num_gpus=0, instascale=False, machine_types=["m5.xlarge", "p3.8xlarge"],
ingress_options = {
    "ingresses": [
        {
            "ingressName": "ray-dashboard-test",
            "port": 8265,
            "pathType": "Prefix",
            "path": "/",
            "host":"<your_host>",
        },
        }
    ]
}))
```
Without specifying any ingress options the default dashboard and client ingresses will be created provided you have set the ingressDomain (note this is only a Kubernetes requirement)

## Important steps for Kind Clusters
Add the `- --enable-ssl-passthrough` argument to the nginx ingress controller
Add the `ingress_domain` field in your cluster configuration to your local domain if you are not using `ingress_options`

# Verification steps



Specify your ingress Options in ClusterConfiguration
Run `cluster.up()` after authenticating in a Jupytr Notebook
Run `kubectl get ingress`
You should get ingresses you specified
upon running `cluster.details()` you should be able to follow the link to the ray dash board provided you created an ingress for it or omitted ingressOptions.
running `cluster.down()` will delete the ingress.
## Checks
- Testing Strategy
